### PR TITLE
chore: open source

### PR DESCRIPTION
**Pull request by ab93**
Open source Python SDK for Numaflow.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Contributing
```
Comment (vigith) [inline CONTRIBUTING.md:1]: can you check what we added in Numaflow? it is pointing to numaproj

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+coverage:
```
Comment (vigith) [inline codecov.yml:1]: do we need this?

File: README.md [hunk 3]
```diff
@@ -1 +1,32 @@
-# numaflow-python
\ No newline at end of file
+# Python SDK
+
+This SDK provides the interface of writing a UDF in Python.
```
Comment (whynowy) [inline README.md:3]: UDSink as well.

**Review by vigith — COMMENTED** [review]
remove `pynumaflow/.DS_Store` and add it in gitignore

also lot of `.pyc` files are coming in

File: pynumaflow/tests/function/.pytest_cache/.gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# Created by pytest automatically.
+v
```
Comment (whynowy) [inline pynumaflow/tests/function/.pytest_cache/.gitignore:2]: Exclude `.DS_Store`

**Comment by vigith** [comment]
also python test cache files are there

File: codecov.yml (hunk 2 above)
Comment (ab93) [reply codecov.yml:1]: Not now. Removing

File: pynumaflow/.gitignore [hunk 5]
```diff
@@ -0,0 +1,35 @@
+*.class
```
Comment (vigith) [inline pynumaflow/.gitignore:1]: this looks like java gitignore.. can you make it python?

File: README.md [hunk 6]
```diff
@@ -1 +1,54 @@
...
```
Comment (vigith) [inline README.md:3]:
```suggestion
This SDK provides the interface for writing [UDFs](https://numaproj.github.io/numaflow/user-defined-functions/) and [UDSinks](https://numaproj.github.io/numaflow/sinks/user-defined-sinks/) in Python.
```

**Review by vigith — APPROVED** [review]
LGTM

**Pull request merged**